Examples PEP for DGD

sebcolla opened the pull request:
Codes used for obtaining the numerical results in the following paper:
Colla, Sebastien, and Julien M. Hendrickx. "Automated Worst-Case Performance Analysis of Decentralized Gradient Descent." (2021)
[https://arxiv.org/abs/2103.14396](url)

sebcolla committed:
Examples PEP for DGD

Codes used for obtaining the results of the following paper:
Colla, Sebastien, and Julien M. Hendrickx. "Automated Worst-Case Performance Analysis of Decentralized Gradient Descent." (2021)

AdrienTaylor reviewed (changes requested):
Hello,
Voici qql petites modifications à apporter pour uniformiser cela dans la toolbox (je n'ai pas encore fait tourner le code, juste regardé):
- mettre les exemples dans le sous-dossier "Examples/11_Decentralized Gradient Method/" (la raison pour laquelle il y a un dossier différent pour CDC, c'est que c'était le dossier original avec les exemples initiaux; je changerai ça un jour. Tes exemples sont facilement classables, alors je préfère le faire; il y en a déjà beaucoup, et sinon ça devient le bordel)
- écrire un script d'exemple par fonction implémentée; avec nom compatible avec les autres exemples, par exemple "A_DGD_PerformanceOnGivenGraph.m" et "B_DGD_PerformanceOnFamilyOfGraphs.m" (noms que tu préfères, mais ça doit être relativement explicite)
- est-ce que tu pourrais changer la ligne 246 en ajoutant le mot "scalar" dans le texte? du genre " 'PESTO: Setting up the problem: other scalar constraints' " (scalar devant constraints)
- SDP -> PSD, et pas très bonne pratique de donner ça comme nom de contrainte dans le code, je crois (vu qu'on veut résoudre un sdp à la fin; tu peux mettre "PSD constraint" par exemple, dans les noms et dans le texte) :-)
- en faisant le changement liste -> cell pour les variables dualles, la liste de noms ("names") n'a pas la même longueur que la liste de variables dualles; tu peux donner un nom à la première? (eg, "Gram matrix is PSD", ou bien "Gram Matrix PSD", ou autre)
- ça doit être le diff qui fait qql chose de bizarre, mais je ne vois pas ce qui a changé à la ligne 331.

Merci!

sebcolla committed:
changes for pull request

sebcolla committed:
add source of the paper for Decentralized Examples

sebcolla committed:
PEP for Decentralized Optimization

New feature for PESTO: possibility of adding new type of constraints (LMIs)
New examples: Decentralized Gradient Descent (DGD)

sebcolla committed:
Update pep output (dualvalues and dualnames)

sebcolla committed:
Update Examples Decentralized Optimization

sebcolla committed:
Update Decentralized Optimization Examples
